lib/dotcom_web/live/trip_planner.ex

```diff
@@ -306,11 +315,12 @@ defmodule DotcomWeb.Live.TripPlanner do
 
     datetime_type = socket.assigns.input_form.changeset.changes.datetime_type
     future = nearest_5_minutes()
+    diff = Timex.diff(datetime, future, :minutes)
 
-    if datetime_type != "now" && Timex.before?(datetime, future) do
+    if datetime_type != "now" && diff < 0 do
       push_event(socket, "set-datetime", %{datetime: future})
     else
-      socket
+      assign(socket, :input_form, Map.put(socket.assigns.input_form, :changeset, Map.put(socket.assigns.input_form.changeset, :datetime, datetime)))
```

Comment:
📓 Looks at the old and new time according to the minute so that it doesn't reset if they are just seconds apart...which will happen if we take `now` two separate times.